src/realm/object-store/c_api/sync.cpp

```diff
@@ -507,41 +507,41 @@ RLM_API void realm_sync_config_set_resync_mode(realm_sync_config_t* config,
 RLM_API realm_object_id_t realm_sync_subscription_id(const realm_flx_sync_subscription_t* subscription) noexcept
 {
     REALM_ASSERT(subscription != nullptr);
-    return to_capi(subscription->id());
+    return to_capi(subscription->id);
 }
 
 RLM_API realm_string_t realm_sync_subscription_name(const realm_flx_sync_subscription_t* subscription) noexcept
 {
     REALM_ASSERT(subscription != nullptr);
-    return to_capi(subscription->name());
+    return to_capi(subscription->name ? "" : *subscription->name);
```

Comment:
I gave up on preserving this behavior. I think it was wrong anyway. The only way to get it is `subscription->name ? to_capi(*subscription->name) : to_capi("")`, and that doesn't seem worth it in order to get the broken old behavior. The returned string probably *should* be null if the subscription has no name.

Reply:
The new code is just `to_capi(subscription->name)` this surprizingly works correctly via the `to_capi(StringData)` overload which in turn works because of the `StringData(const optional<std::string>&)` implicit conversion.

I don't see how the old code provided a way to distinguish errors from unset strings.

Reply:
The current version of this line is here fwiw https://github.com/realm/realm-core/pull/6065/files#diff-ba7e9e5a00e04033638ced899e7f7f613243e4e2685fbdf3c519798b4e711d35R516

Reply:
Yep, I think it is fine like it is... I would not check the pointer in there, the most important thing is that we don't end up hitting another ctor by mistake in the future and crash because of a subscription that does not have a name :-)